### DISCUSSION TITLE
how to import on startup?  xonsh doesnt respect PYTHONSTARTUP (solved)

### DISCUSSION THREAD

**Discussion by shula** [xonsh/xonsh#5651] [General]
(SOLVED - i'll keep to save time for others like me who couldn't easily find it)

I use many imports (e.g. from module import func) on my daily sessions.
I'd like to `import` them to each shell i start + initialize some variables.

With current python interpreter, the PYTHONSTARTUP system var points to a python script, which does the above. But xonsh doesn't respect that (why?).

example of such python_startup.py:
```py
import this, that
txt = 'lorry M. possum'
outfile = open('/tmp/out','a')
print(' -- this, that, outfile, and txt are available')
```

(there's a slight performance penalty of course)

The solution:
1) look at $XONSHRC_DIR *
2) create these folders if they don't exist    **
2) create or link your startup script in one of these are the folders 

Next time you start xonsh, "this" and "that" will be available

* (run xonsh, and just write `$XONSHRC_DIR` on the prompt).
** (suggestion:  create these folders on install... )
(security measures apply like any other .bashrc)

**Comment by anki-code** [comment]
Thanks for sharing this!

> With current python interpreter, the PYTHONSTARTUP system var points to a python script, which does the above. But xonsh doesn't respect that (why?).

It's just a new feature :)

> I'd like to import them to each shell i start + initialize some variables.

Just to let you know that you can use lazy loading to reduce strartup time. Take a look into `xonsh.lib.lazy*` or search for "lazy loading in python".
